### PR TITLE
Feature/61 signin menu

### PR DESCRIPTION
**Pull request by SiboYang**
# Summary

Created a dropdown menu and sign in button

## Test Plan

No test

## Related Issues

Which issues does this PR resolve/work on? 
#61

### CODE REVIEW

File: frontend/src/DropdownMenu/DropdownMenu.css [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.DropdownMenu {
+  position: absolute;
+  top: 60px;
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.css:3]: This seems to work well! Now let's use [makeStyles](https://material-ui.com/styles/basics/#hook-api) and Material UI's spacing components in DropdownMenu.tsx with whole-number margins (see [here](https://material-ui.com/system/spacing/)). This allows for margins to be parametrized and changed globally later.

File: frontend/src/DropdownMenu/DropdownMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        color="primary"
+        onClick={handleClick}
+      >
+        More
+      </Button>
+      <Menu
+        id="simple-menu"
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.tsx:33]: Use the [double bang](https://medium.com/dev-genius/vs-boolean-cd502fbd6101), `!!anchorEl`

File: frontend/src/DropdownMenu/DropdownMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        onClose={handleClose}
+      >
+        <MenuItem>
+          <TranslateIcon /> English (Anglais)
+        </MenuItem>
+        <MenuItem>Turn Location Off</MenuItem>
+        <MenuItem>Visit MU site</MenuItem>
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.tsx:40]: Might as well add `component="a" href="https://mumtl.org/"`

File: frontend/src/SigninButton/SigninButton.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.SigninButton {
+  position: absolute;
+  top: 10px;
```
Comment (tedspare) [inline frontend/src/SigninButton/SigninButton.css:3]: Same as above - replace this CSS with `makeStyles`

File: frontend/src/SigninButton/SigninButton.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+interface ISigninButtonProps {
+  isSignedIn: boolean;
+  onClick: () => void;
+}
+
+function SigninButton(props: ISigninButtonProps) {
+  return (
+    <div className="SigninButton">
+      {props.isSignedIn ? (
```
Comment (tedspare) [inline frontend/src/SigninButton/SigninButton.tsx:13]: What about making the ternary expression `a ? b : c` finer-grained? For example, "SIGN IN" and "SIGN OUT" could be extracted to strings in a map then we would use `...>props.isSignedIn ? buttonText.signOut : buttonText.signIn<...`

**Review by tedspare — APPROVED** [review]
Added a few comments otherwise looks good to me!

File: frontend/src/App/App.tsx [hunk 6]
```diff
@@ -1,11 +1,20 @@
...
 function App() {
+  const [isSignedIn, setIsSignedIn] = useState<boolean>(false);
+  function handleSigninClick() {
+    setIsSignedIn(!isSignedIn);
+  }
+
   return (
     <div className="App">
       <Map mapContainer={document.getElementById("root")} />
+      <SigninButton isSignedIn={isSignedIn} onClick={handleSigninClick} />
```
Comment (tedspare) [inline frontend/src/App/App.tsx:16]: What do you think about placing the SigninButton inside the DropdownMenu?

File: frontend/src/DropdownMenu/DropdownMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  return (
+    <div className="DropdownMenu">
+      <Button
+        aria-controls="simple-menu"
+        aria-haspopup="true"
+        color="primary"
+        onClick={handleClick}
+      >
+        More
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.tsx:27]: What if we replace this text with a MenuIcon?

Comment (SiboYang) [reply frontend/src/DropdownMenu/DropdownMenu.tsx:27]: Thanks for the comments,  I will fix them now!

Comment (SiboYang) [reply frontend/src/DropdownMenu/DropdownMenu.tsx]: Hi Ted, I modified the code, not sure if that follow you suggestions, please correct me if I am wrong!

File: frontend/src/DropdownMenu/DropdownMenu.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const buttonText = {
+    signin: <strong style={{ color: "#216ADB" }}>SIGN IN</strong>,
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.tsx:34]: Let's import and use `constants.PRIMARY` and `constants.SECONDARY`

File: frontend/src/DropdownMenu/DropdownMenu.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+interface IDropdownMenuProps {
+  isSignedIn: boolean;
+  signInClick: () => void;
+}
+
+const useStyles = makeStyles({
+  DropdownMenu: {
+    position: "absolute",
+    top: "10px",
```
Comment (tedspare) [inline frontend/src/DropdownMenu/DropdownMenu.tsx:17]: Use

```
import { createMuiTheme } from "@material-ui/core/styles";
const theme = createMuiTheme();
```

then `theme.spacing(1);`

File: frontend/src/DropdownMenu/DropdownMenu.tsx (hunk 7 above)
Comment (tedspare) [reply frontend/src/DropdownMenu/DropdownMenu.tsx]: I think your solution is generally very clean, readable, and functional across mobile and web  🙌  
Thanks for incorporating the feedback.

**Pull request merged**